### PR TITLE
Carthage example for EarlGrey 2

### PR DESCRIPTION
**Pull request by bootstraponline**

### CODE REVIEW

File: Demo/CarthageExample/CarthageExample/ViewController.swift [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import UIKit
+
+class ViewController: UIViewController {
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+    }
+
```
Comment (tirodkar) [inline Demo/CarthageExample/CarthageExample/ViewController.swift]: rm line.

**Comment by bootstraponline** [comment]
> While this seems fine. My question was - is there no way to use the existing EarlGrey 2.0 functional tests rig?

Discussed on slack to rename to `DemoApp` with targets for Carthage and any other build systems we add in the future.

After updating EarlGrey to ad70b511da47793e27e77c1a69c7895e8403be45 the example no longer compiles.

**Comment by bootstraponline** [comment]
I updated with the requested changes. The code compiles now with no changes. Flaky build technology is fun.

image

File: Demo/DemoApp/DemoApp/Assets.xcassets/AppIcon.appiconset/Contents.json [hunk 2]
```diff
@@ -0,0 +1,98 @@
+{
```
Comment (bootstraponline) [inline Demo/DemoApp/DemoApp/Assets.xcassets/AppIcon.appiconset/Contents.json]: I kept this around to make the DemoApp fully decoupled from the existing test rig (per discussion on Slack).

File: Demo/DemoApp/Cartfile.resolved [hunk 3]
```diff
@@ -0,0 +1 @@
+github "google/EarlGrey" "ad70b511da47793e27e77c1a69c7895e8403be45"
```
Comment (tirodkar) [inline Demo/DemoApp/Cartfile.resolved]: Pinned to a commit?

Comment (bootstraponline) [reply Demo/DemoApp/Cartfile.resolved]: `carthage update` will refresh the `Cartfile.resolved` file with the latest commit

`Cartfile` is listed as tracking earlgrey2 branch

**Comment by tirodkar** [comment]
LGTM.

**Comment by tirodkar** [comment]
Want to add a target for this on Bitrise as well?

**Comment by bootstraponline** [comment]
> Want to add a target for this on Bitrise as well?

That's the plan. I'll update Bitrise after merge. Bitrise isn't setup to build forks

File: Demo/DemoApp/DemoApp.xcodeproj/project.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist [hunk 4]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (khandpur) [inline Demo/DemoApp/DemoApp.xcodeproj/project.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist]: just wanted to check if all these IDE related files are required or will Xcode self-generate them if they don't exist.

File: Demo/DemoApp/readme.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
+# EarlGrey 2 + Carthage
```
Comment (khandpur) [inline Demo/DemoApp/readme.md]: Nit: I'd capitalize the file name to `README.md`

File: Demo/DemoApp/DemoApp.xcodeproj/project.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist (hunk 4 above)
Comment (bootstraponline) [reply Demo/DemoApp/DemoApp.xcodeproj/project.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist]: That's a great question. I think we want to keep this file.

> Xcode 9.3 adds a new IDEWorkspaceChecks.plist file to a workspace's shared data, to store the state of necessary workspace checks. Committing this file to source control will prevent unnecessary rerunning of those checks for each user opening the workspace. (37293167)
>
> https://stackoverflow.com/questions/50367896/ideworkspacechecks-plist-file-suddenly-appear-after-updated-xcode

File: Demo/DemoApp/readme.md (hunk 5 above)
Comment (bootstraponline) [reply Demo/DemoApp/readme.md]: Done!

File: Demo/DemoApp/DemoApp.xcodeproj/project.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist (hunk 4 above)
Comment (khandpur) [reply Demo/DemoApp/DemoApp.xcodeproj/project.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist]: awesome. Thanks for checking!

**Pull request merged**